lib/pacemaker/pcmk_output.c

```diff
@@ -372,35 +372,33 @@ stacks_and_constraints_xml(pcmk__output_t *out, va_list args) {
     return pcmk_rc_ok;
 }
 
-PCMK__OUTPUT_ARGS("health", "const char *", "const char *", "const char *", "const char *")
+PCMK__OUTPUT_ARGS("health", "const char *", "const char *", "const char *")
 static int
 health_text(pcmk__output_t *out, va_list args)
 {
-    const char *sys_from = va_arg(args, const char *);
     const char *host_from = va_arg(args, const char *);
     const char *fsa_state = va_arg(args, const char *);
     const char *result = va_arg(args, const char *);
 
     if (!out->is_quiet(out)) {
-        out->info(out, "Status of %s@%s: %s (%s)", crm_str(sys_from),
-                       crm_str(host_from), crm_str(fsa_state), crm_str(result));
+        out->info(out, "Controller on %s in state %s: %s", crm_str(host_from),
+                        crm_str(fsa_state), crm_str(result));
     } else if (fsa_state != NULL) {
         out->info(out, "%s", fsa_state);
     }
 
     return pcmk_rc_ok;
 }
 
-PCMK__OUTPUT_ARGS("health", "const char *", "const char *", "const char *", "const char *")
+PCMK__OUTPUT_ARGS("health", "const char *", "const char *", "const char *")
 static int
 health_xml(pcmk__output_t *out, va_list args)
 {
-    const char *sys_from = va_arg(args, const char *);
     const char *host_from = va_arg(args, const char *);
     const char *fsa_state = va_arg(args, const char *);
     const char *result = va_arg(args, const char *);
 
-    pcmk__output_create_xml_node(out, crm_str(sys_from),
+    pcmk__output_create_xml_node(out, "Controller",
```

Comment:
XML is different :) Since it's intended for machine parsing rather than humans, the IPC name is appropriate. Also, this is used as the tag name, so we'd have to change the schema if we changed it. That does mean the text function should accept the argument, just mark it G_GNUC_UNUSED